src/lazy/text/matched.rs

```diff
@@ -634,6 +672,57 @@ fn decode_hex_digits_escape<'data>(
     Ok(remaining_input)
 }
 
+/// Reads another escaped code point from the buffer, treating it as the low surrogate to be paired
+/// with the specified high surrogate. Appends the UTF-8 encoding of the resulting Unicode scalar
+/// to `sanitized` and returns the remaining text in the buffer.
+fn complete_surrogate_pair<'data>(
```

Comment:
🗺️ I added support for these to pass the `ion-tests` around strings/symbols. See amazon-ion/ion-tests#637